### PR TITLE
Migrate validate dialog to Lit

### PR DESCRIPTION
**Pull request by balloob**
Migrate the validate dialog and the foundation (log modal) to Lit.

Unblocks other migrations mentioned in #4.

### CODE REVIEW

File: src/components/remote-process.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  public connectedCallback() {
+    if (this._setup) {
+      return;
+    }
+    this._setup = true;
+    const shadowRoot = this.attachShadow({ mode: "open" });
+
+    shadowRoot.innerHTML = `
+      <style>
+        .log {
```
Comment (balloob) [inline src/components/remote-process.ts:23]: This is copied from esphome.css. Can be removed from there once all log-based dialogs are migrated.

File: src/components/remote-process.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+@customElement("esphome-remote-process")
+class ESPHomeRemoteProcess extends HTMLElement {
+  public type!: "validate" | "logs" | "upload";
+  public filename!: string;
+
+  private _abortController?: AbortController;
+  private _setup = false;
+
+  public connectedCallback() {
+    if (this._setup) {
```
Comment (bramkragten) [inline src/components/remote-process.ts:15]: Shouldn't we restart what we aborted if we connect again?

File: src/install-update/index.ts [hunk 3]
```diff
@@ -2,10 +2,10 @@ import { uploadModal } from "../legacy";
 
 const preload = () => import("./install-update-dialog");
 
-const startInstallUpdate = async (ev: Event) => {
+export const openInstallDialog = async (filename: string) => {
   preload();
   const dialog = document.createElement("esphome-install-dialog");
```
Comment (bramkragten) [inline src/install-update/index.ts:7]: Do we have to create a new dialog element every time, or should we re-use it? Do we also remove the dialog when we close it?

File: src/validate/validate-dialog.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+  private _openInstall() {
+    openInstallDialog(this.filename);
+  }
+
+  private _handleProcessDone(ev: { detail: number }) {
+    this._valid = ev.detail == 0;
+  }
+
+  private async _handleClose() {
```
Comment (bramkragten) [inline src/validate/validate-dialog.ts:67]:
```suggestion
  private _handleClose() {
```

**Review by bramkragten — APPROVED** [review]

File: src/install-update/index.ts (hunk 3 above)
Comment (bramkragten) [reply src/install-update/index.ts:7]: Yes we do.

File: src/components/remote-process.ts (hunk 2 above)
Comment (balloob) [reply src/components/remote-process.ts:15]: Nah we should create a new process

**Pull request merged**